modules/transactionpool/transactionpool.go

```diff
@@ -58,19 +63,17 @@ type (
 )
 
 // New creates a transaction pool that is ready to receive transactions.
-func New(cs modules.ConsensusSet, g modules.Gateway) (tp *TransactionPool, err error) {
+func New(cs modules.ConsensusSet, g modules.Gateway) (*TransactionPool, error) {
 	// Check that the input modules are non-nil.
 	if cs == nil {
-		err = errors.New("transaction pool cannot use a nil state")
-		return
+		return nil, errNilCS
 	}
 	if g == nil {
-		err = errors.New("transaction pool cannot use a nil gateway")
-		return
+		return nil, errNilGateway
```

Comment:
too bad :)

I think it does pay off to be meticulous. Just have to be meticulous in the right places. I do feel like 100% code coverage is one of those places.